### ISSUE TITLE
E3032 for keyschema in dynamodb table false positive

### ISSUE THREAD

**Issue by AyoubOukh** [aws-cloudformation/cfn-lint#3619]
### CloudFormation Lint Version

1.10.3

### What operating system are you using?

ubuntu

### Describe the bug

```
E3032 [{'AttributeName': {'Ref': 'HashKeyName'}, 'KeyType': 'HASH'}] is too short (2)
Error: table-template.yml:215:7
```
 the e3032 is raised even if there is the minimum required hash key for keyschema in dynamodb table

### Expected behavior

It shouldn't fail for e3032

### Reproduction template

```
  Table:
    Type: AWS::DynamoDB::Table
    Properties:
      TableName: !Ref TableName
      BillingMode: !Ref BillingMode
      AttributeDefinitions:
        - AttributeName: !Ref HashKeyName
          AttributeType: !Ref HashKeyType
        - !If
          - HasRangeKey
          - AttributeName: !Ref RangeKeyName
            AttributeType: !Ref RangeKeyType
          - Ref: AWS::NoValue
        - !If
          - HasAttribute0
          - AttributeName: !Ref Attribute0Name
            AttributeType: !Ref Attribute0Type
          - Ref: AWS::NoValue
      KeySchema:
        - AttributeName: !Ref HashKeyName
          KeyType: HASH
        - !If
          - HasRangeKey
          - AttributeName: !Ref RangeKeyName
            KeyType: RANGE
          - Ref: AWS::NoValue
 ...
```

**Comment by kddejong** [comment]
Can you provide the condition logic for `HasRangeKey` also are you using [LocalSecondaryIndexes](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-localsecondaryindexes) in your resource?

**Comment by AyoubOukh** [comment]
```
Conditions:
  HasRangeKey: !Not [!Equals [!Ref RangeKeyName, '']]
  HasLocalSecondaryIndex0: !Not [!Equals [!Ref LocalSecondaryIndex0Name, '']]
  HasLocalSecondaryIndex1: !Not [!Equals [!Ref LocalSecondaryIndex1Name, '']]
```
Yes we do have them but also on a condition:
```
      LocalSecondaryIndexes:
        - !If
          - HasLocalSecondaryIndex0
          - IndexName: !Ref LocalSecondaryIndex0Name
            KeySchema:
              - AttributeName: !Ref HashKeyName
                KeyType: HASH
              - AttributeName: !Ref LocalSecondaryIndex0RangeKey
                KeyType: RANGE
            Projection:
              ProjectionType: ALL
          - Ref: AWS::NoValue
        - !If
          - HasLocalSecondaryIndex1
          - IndexName: !Ref LocalSecondaryIndex1Name
            KeySchema:
              - AttributeName: !Ref HashKeyName
                KeyType: HASH
              - AttributeName: !Ref LocalSecondaryIndex1RangeKey
                KeyType: RANGE
            Projection:
              ProjectionType: ALL
          - Ref: AWS::NoValue
```
If I rely solely on the table deployment success, then I don't think it's a valid error.

**Comment by AyoubOukh** [comment]
Thank you!
